proxy/http/HttpConfig.cc

```diff
@@ -900,6 +900,68 @@ register_stat_callbacks()
                      (int)http_sm_finish_time_stat, RecRawStatSyncSum);
 }
 
+static bool
+set_negative_caching_list(const char *name, RecDataT dtype, RecData data, HttpConfigParams *c, bool update)
+{
+  bool ret = false;
+  // values from proxy.config.http.negative_caching_list
+  if (0 == strcasecmp("proxy.config.http.negative_caching_list", name) && RECD_STRING == dtype) {
+    if (data.rec_string) {
+      // parse the list of status code
+      ts::TextView status_list(std::string(data.rec_string));
+      while (!status_list.empty()) {
+        auto is_sep{[](char c) { return isspace(c) || ',' == c || ';' == c; }};
+        ts::TextView span, token{status_list.ltrim_if(is_sep).take_prefix_if(is_sep)};
+        auto n = ts::svtoi(token, &span);
+        if (span.size() != token.size()) {
+          Error("Invalid status code '%.*s' for negative caching: not a number", static_cast<int>(token.size()), token.data());
+        } else if (n <= 0 || n >= 600) {
+          Error("Invalid status code '%.*s' for negative caching: out of range", static_cast<int>(token.size()), token.data());
+        } else {
+          c->negative_caching_list[n] = true;
+        }
+      }
+      ret = true;
+    }
+  }
+  if (update == true) {
```

Comment:
Perhaps `return update`? Or assign to `zret` instead of `ret` or `update`. There's no comment indicating what exactly the return value of this function means so it's hard to be sure.